src/scmdata/run.py

```diff
@@ -1614,13 +1640,31 @@ def process_over(
         else:
             res = grouper.apply(operation, **kwargs)
 
+        if op_cols is not None:
+            idx_df = res.index.to_frame()
+            for column_name in op_cols:
+                idx_df[column_name] = op_cols[column_name]
+            res.index = pd.MultiIndex.from_frame(idx_df)
+
         if na_override is not None:
             idx_df = res.index.to_frame()
             idx_df[idx_df == na_override] = np.nan
             res.index = pd.MultiIndex.from_frame(idx_df)
 
         res = res.reorder_levels(sorted(res.index.names))
-        return res
+
+        if as_run:
+            try:
+                if issubclass(as_run, BaseScmRun):
+                    Cls = as_run
+                else:
+                    Cls = self.__class__
+            except TypeError:
```

Comment:
Seems safer to check if `as_run` is a boolean rather than having the type error silently pass?

Reply:
Maybe a test of the error you get if `as_run` is not a subclass of `BaseScmRun`? If you put `as_run=pd.DataFrame`, I think the behaviour wouldn't be entirely obvious in the current setup?

Reply:
Added